xbto/test/adapter.test.ts

```diff
@@ -0,0 +1,45 @@
+import { assert } from 'chai'
+import { Requester, assertSuccess, assertError } from '@chainlink/external-adapter'
+import { AdapterRequest } from '@chainlink/types'
+import { execute } from '../src/adapter'
+
+describe('execute', () => {
+  const jobID = '1'
+
+  context('successful calls @integration', () => {
+    const requests = [
+      {
+        name: 'id not supplied',
+        testData: { data: {} },
+      },
+      {
+        name: 'id supplied',
+        testData: { id: jobID, data: {} },
+      },
+    ]
+
+    requests.forEach((req) => {
+      it(`${req.name}`, async () => {
+        const data = await execute(req.testData as AdapterRequest)
+        assertSuccess({ expected: 200, actual: data.statusCode }, data, jobID)
+        assert.isAbove(data.result, 0)
+        assert.isAbove(data.data.result, 0)
+      })
+    })
+  })
+
+  context('validation error', () => {
+    const requests = [{ name: 'empty body', testData: {} }]
+
+    requests.forEach((req) => {
+      it(`${req.name}`, async () => {
+        try {
+          await execute(req.testData as AdapterRequest)
+        } catch (error) {
+          const errorResp = Requester.errored(jobID, error)
+          assertError({ expected: 400, actual: errorResp.statusCode }, errorResp, jobID)
+        }
+      })
+    })
+  })
```

Comment:
```suggestion
```
I believe what is happening is this the empty body test returns a `200` because there are no params, so when asserting that the error should be  `400` it retries 3 times and then times out returning a `500` error. That's why it hits the timeout, because to do that takes about 4-5 seconds.

Moving this test to "successful calls" makes the unit tests pass. Or the other option is increasing the timeout in `package.json` **AND** changing this to assert for a `500` error works, but would be just getting around something that should be fixed in the test suite.